### DISCUSSION TITLE
Constraint set of routes, based on current connected account

### DISCUSSION THREAD

**Discussion by bdavidxyz** [janko/rodauth-rails#93] [Q&A] [answered]
With Devise, you can write something like this : 
```
# config/routes.rb
Rails.application.routes.draw do
  authenticate :user, ->(user) { user.admin? } do
    # ...
  end
end
```

With Rodauth-Rails, you can write something like this : 

```
# config/routes.rb
Rails.application.routes.draw do
  constraints -> (r) { r.env["rodauth"].logged_in? } do 
    #...
  end
end
```

Is there a way to access the current_account in the constraints ? Not only the "rodauth" object ? `current_account` seem to be available with Views And Controllers only...

Thanks

**Comment by renchap** [comment, marked as answer]
You can do it this way:

```ruby
 constraints(
    Rodauth::Rails.authenticated do |rodauth|
      User.find(rodauth.session_value)&.admin?
    end,
  ) do
    # ...
  end
```

**Comment by janko** [comment]
Thanks @renchap for posting a solution. As of version 1.2, you can now retrieve the model instance via the `#rails_account` Rodauth method:

```rb
constraints Rodauth::Rails.authenticated { |rodauth| rodauth.rails_account.admin? } do
  # ...
end
```

This has the benefit of not re-fetching the account record when `#current_account` is called in the same request.
